test/hotspot/jtreg/compiler/c2/TestBitCompressValueTransform.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
```

Comment:
> Ah, I just noticed the test directory. I think we can put it in a more specific location.

> There are operation specific tests under compiler/c2 let keep it this way.

But this is about a Value optimization, which generally belong under the gvn directory. Just because we used to put all tests in one directory does not mean we should continue that practice ;)

We should probably at some point clean up all the tests, and move them to better directories. Putting them already to better directories will reduce the backport issues later on, so please put it somewhere more specific.

@jatin-bhateja I'm not very happy when people just "resolve" a conversation unilaterally. Can you please avoid that in the future?